From e772270774ff45172709e39f744cab875a816667 Mon Sep 17 00:00:00 2001
From: Artturin <Artturin@artturin.com>
Date: Sun, 11 Sep 2022 23:29:58 +0300
Subject: [PATCH] rpcgen: follow the CPP and RPCGEN_CPP enviromental variables

cpp may be target prefixed
and it may cause issues in other programs if CPP is set
---
 rpcgen/rpc_main.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/rpcgen/rpc_main.c b/rpcgen/rpc_main.c
index 277adc6..1d0720d 100644
--- a/rpcgen/rpc_main.c
+++ b/rpcgen/rpc_main.c
@@ -348,6 +348,12 @@ find_cpp (void)
       crash ();
     }
 
+  if ((CPP = getenv("RPCGEN_CPP")) != NULL)
+    return;
+
+  if ((CPP = getenv("CPP")) != NULL)
+    return;
+
   /* fall back to system CPP */
   CPP = "cpp";
 }
-- 
2.50.1 (Apple Git-155)

